Libraries/Vibration/Vibration.js

```
@@ -27,11 +27,31 @@ var Platform = require('Platform');
  */
 
 var Vibration = {
-  vibrate: function(duration: number = 400) {
+  vibrate: function(pattern: number | Array<number> = 400, repeat: number = -1) {
     if (Platform.OS === 'android') {
-      RCTVibration.vibrate(duration);
+      if (typeof pattern === 'number') {
+        RCTVibration.vibrate(pattern);
+      } else if (Array.isArray(pattern)) {
+        RCTVibration.vibrateByPattern(pattern, repeat);
+      }
```

If pattern is not a number or array, let's throw an error.